src/utils/graph.ts

```diff
@@ -49,22 +61,27 @@ export const addEdge = (edgeParams: Edge | Connection, elements: Elements): Elem
 
   // make sure that there is node with the target and one with the source id
   [edgeParams.source, edgeParams.target].forEach((id) => {
-    const nodeId = id.includes('__') ? id.split('__')[0] : id;
-    if (!elements.find((e) => isNode(e) && e.id === nodeId)) {
-      throw new Error(`Can't create edge. Node with id=${nodeId} does not exist.`);
+    if (!elements.find((e) => isNode(e) && e.id === id)) {
+      throw new Error(`Can't create edge. Node with id=${id} does not exist.`);
     }
   });
+  // need to check if handles exist
```

Comment:
hello @moklick , I just made the changes you requested. The fork I'm working on right now is 6.1.3 with both my pull requests integrated. Should I remove my current pull requests and submit a new one? #543 is from version 6.0.0 and #568 is from version 6.1.3 but does not have #543 changes. It's confusing and you might have a hard time merging changes.

Reply:
For clarification, my current fork has:
- The handles update associated with #543 with the fixes you requested. I also modified the Validation example to showcase the usefulness of breaking out handles by itself.
- The drag and drop changes associated with #569 along with a new example to show how it can be used.

[link to my current fork](https://github.com/jasonpul/react-flow)

Reply:
The problem right now is that #568 also contains #543 when I see it right. It would be necessary that you divide the changes so that we can merge them separately.

Reply:
ok i'll break them out and resubmit. Thanks!